sdk/eventhubs/azure-messaging-eventhubs/test/ut/checkpoint_store_test.cpp

```diff
@@ -46,7 +46,7 @@ namespace Azure { namespace Messaging { namespace EventHubs { namespace Test {
 
     checkpoints = checkpointStore.ListCheckpoints(
         "ns.servicebus.windows.net", "event-hub-name", "$Default");
-    EXPECT_EQ(checkpoints.size(), 1);
+    EXPECT_EQ(checkpoints.size(), 1ul);
```

Comment:
Should we also fix other recent check-ins with `vector.size()` being compared to non-unsigned values, like a couple in this?
https://github.com/Azure/azure-sdk-for-cpp/blob/972c3d8c76865c538d2050df5b24d2d391bbc289/sdk/core/azure-core-amqp/test/ut/amqp_value_tests.cpp#L1251

Reply:
If the build isn't broken and the tests aren't failing, I don't feel strongly about fixing these.

I care about changes that break the build, which is what this PR is all about.

Reply:
Unless I am missing something, since the root cause is the same, I would have expected the instances in amqp to also break the build. Are we building eventhubs and amqp tests in different build configurations?

Reply:
You're absolutely right - the root cause is the same.

In this case, the ONLY thing that matters here is whether or not the nightly build is broken or not. 

I don't know why the x86 compiler is complaining about the pattern in eventhubs but not in amqp. But it is not complaining about amqp and it is complaining about eventhubs. Also, VS 2022 doesn't complain but VS 2019 does.

Since this PR is about fixing the nightly build, I only want to fix the nightly build. I'm not interested in fixing code which isn't currently broken.